### PR TITLE
[MSHARED-1043] Deprecate methods leveraging JUnit4 assertions

### PR DESCRIPTION
**Pull request by kwin**
Provide alternatives relying on framework agnostic exception instead

### CODE REVIEW

**Review by elharo — COMMENTED** [review]
some nits

**Review by slawekjaranowski — COMMENTED** [review]
It will be appreciated to add javadocs for new public methods. 
Especially that you link new methods in deprecated ones.

File: src/main/java/org/apache/maven/it/Verifier.java [hunk 1]
```diff
@@ -1020,66 +1036,146 @@ public void assertFilePresent( String file )
...
-            catch ( VerificationException e )
-            {
-                Assert.fail( e.getMessage() );
-            }
+            verifyFilePresence( fileName, wanted );
         }
     }
 
+    /**
+     * Verifies that the artifact given through its Maven coordinates exists.
```
Comment (elharo) [inline src/main/java/org/apache/maven/it/Verifier.java:1124]: . \n

File: src/main/java/org/apache/maven/it/Verifier.java [hunk 2]
```diff
@@ -1020,66 +1036,146 @@ public void assertFilePresent( String file )
...
+     * @param groupId the groupId of the artifact (must not be null)
+     * @param artifactId the artifactId of the artifact (must not be null)
+     * @param version the version of the artifact (must not be null)
+     * @param ext the extension of the artifact (must not be null)
+     * @throws VerificationException in case the given artifact does not exist
```
Comment (elharo) [inline src/main/java/org/apache/maven/it/Verifier.java:1129]: in case --> if

File: src/main/java/org/apache/maven/it/Verifier.java [hunk 3]
```diff
@@ -1020,66 +1036,146 @@ public void assertFilePresent( String file )
...
+        verifyArtifactPresence( true, groupId, artifactId, version, ext );
+    }
+
+    /**
+     * Verifies that the artifact given through its Maven coordinates does not exist.
+     * @param groupId the groupId of the artifact (must not be null)
+     * @param artifactId the artifactId of the artifact (must not be null)
+     * @param version the version of the artifact (must not be null)
+     * @param ext the extension of the artifact (must not be null)
+     * @throws VerificationException in case the given artifact exists
```
Comment (elharo) [inline src/main/java/org/apache/maven/it/Verifier.java:1143]: in case --> if

File: src/main/java/org/apache/maven/it/Verifier.java [hunk 4]
```diff
@@ -1020,66 +1036,146 @@ public void assertFilePresent( String file )
...
+        catch ( VerificationException e )
+        {
+            Assert.fail( e.getMessage() );
+        }
+    }
+
+    /**
+     * @deprecated Use {@link #verifyArtifactPresent(String, String, String, String)} instead.
+     */
+    @Deprecated
```
Comment (elharo) [inline src/main/java/org/apache/maven/it/Verifier.java:1166]: I guess we do. Use this elsewhere where needed.

File: src/main/java/org/apache/maven/it/Verifier.java [hunk 5]
```diff
@@ -1768,6 +1864,29 @@ private static void runIntegrationTest( Verifier verifier )
         System.out.println( "OK" );
     }
 
+    /**
+     * Verifies that the artifact given by its Maven coordinates exists and contains the given content.
```
Comment (elharo) [inline src/main/java/org/apache/maven/it/Verifier.java:1868]: blank line at end

File: src/main/java/org/apache/maven/it/Verifier.java [hunk 6]
```diff
@@ -1768,6 +1864,29 @@ private static void runIntegrationTest( Verifier verifier )
...
+     * @param groupId the groupId of the artifact (must not be null)
+     * @param artifactId the artifactId of the artifact (must not be null)
+     * @param version the version of the artifact (must not be null)
+     * @param ext the extension of the artifact (must not be null)
+     * @param content the expected content
+     * @throws IOException in case reading from the artifact fails 
+     * @throws VerificationException in case the content of the artifact differs
```
Comment (elharo) [inline src/main/java/org/apache/maven/it/Verifier.java:1875]: if

**Review by elharo — COMMENTED** [review]
I like this idea overall. I'm a little concerned that this is somewhat incompatible with existing usage. What other code outside this project uses the Verifier class?

**Comment by kwin** [comment]
> I'm a little concerned that this is somewhat incompatible with existing usage. 

This code should be fully backwards-compatible as no existing methods are removed/modified and only new ones are added. Some consumers can be seen at https://mvnrepository.com/artifact/org.apache.maven.shared/maven-verifier/usages. In any case a migration away from the deprecated methods should be pretty straightforward.

**Comment by elharo** [comment]
It's more of a long term concern. Is anyone who's not the Maven project using this? If so, it's going to be really hard to remove the deprecated methods. 

I agree that if we had a green field this approach is preferable.

**Comment by slawekjaranowski** [comment]
It is standard way, first deprecated method with documentation how should be replace - and we do this step.

Next we release major version with removed methods in some time.

**Comment by elharo** [comment]
It's the standard way, assuming step 2 can in fact be done. Far too often, step 2 never happens and the deprecated code is never removed and supported forever.

**Comment by elharo** [comment]
Thanks for the link to mvnrepository. That's more usages than I would have guessed. My gut is that we're not going to remove these methods. Too many projects depend on them already. I think we're better off living with what we have.

**Comment by slawekjaranowski** [comment]
Better stats is usage per version, like:
https://mvnrepository.com/artifact/org.apache.maven.shared/maven-verifier

**Review by slawekjaranowski — COMMENTED** [review]
please use deprecated javadoc tag and annotation together.

File: src/main/java/org/apache/maven/it/Verifier.java [hunk 7]
```diff
@@ -1768,6 +1869,30 @@ private static void runIntegrationTest( Verifier verifier )
...
+        String fileName = getArtifactPath( groupId, artifactId, version, ext );
+        if ( content.equals( FileUtils.fileRead( fileName ) ) )
+        {
+            throw new VerificationException( "Content of " + fileName + " does not equal " + content );
+        }
+    }
+
+    /**
+     * @deprecated Use {@link #verifyArtifactContent(String, String, String, String, String)} instead.
+     */
```
Comment (slawekjaranowski) [inline src/main/java/org/apache/maven/it/Verifier.java:1895]: missing deprecated annotation 😉

**Comment by slawekjaranowski** [comment]
@kwin - what with open conversations with @elharo

**Comment by kwin** [comment]
> what with open conversations with @elharo

I think everything should be resolved now.

**Review by slawekjaranowski — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #150

**Pull request merged**